gpumon/gpumon_interface.ml

```diff
@@ -0,0 +1,28 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+let service_name = "gpu"
+let queue_name = Xcp_service.common_prefix ^ service_name
+let json_path = "/var/xapi/" ^ service_name ^ ".json"
+let xml_path = "/var/xapi/" ^ service_name
+
+type debug_info = string
+
+type pgpu_address = string
+type pgpu_metadata = string
+type dom_id = int
+type vgpu_compatibility = unit (* temporarily unit, until real type is determined *)
+
+external get_pgpu_metadata: debug_info -> pgpu_address -> pgpu_metadata = ""
+external get_pgpu_vm_compatibility: debug_info -> pgpu_metadata -> dom_id -> vgpu_compatibility list = ""
```

Comment:
For specifying the PCI device, I had imagined we might be able to use the xapi database's `pci_id` field of a `pci` db entry, provided by xapi with code like `Db.PCI.get_pci_id ~__context ~self:pci ` so it would be of format `"%04x:%02x:%02x.%01x"`
See http://wiki.xen.org/wiki/Bus:Device.Function_%28BDF%29_Notation
and/or `xen-api.git/ocaml/xapi/pciops.ml`